hickle/helpers.py

```diff
@@ -1,141 +1,238 @@
+
+    def __getattribute__(self, name):
+        # for attrs and wrapped _h5_node return local copy any other request
+        # redirect to wrapped _h5_node
+        if name in {"attrs", "_h5_node"}:
+            return super(H5NodeFilterProxy,self).__getattribute__(name)
+        _h5_node = super(H5NodeFilterProxy,self).__getattribute__('_h5_node')
+        return getattr(_h5_node, name)
+        
+    def __setattr__(self, name, value):
+        # if wrapped _h5_node and attrs shall be set store value on local attributes
+        # otherwise pass on to wrapped _h5_node
+        if name in {'_h5_node'}:
+            super().__setattr__(name, value)
+            return
+        if name in {'attrs'}: # pragma: nocover
+            raise AttributeError('attribute is read-only')
+        _h5_node = super().__getattribute__('_h5_node')
+        setattr(_h5_node, name, value)    
+
+    def __getitem__(self, *args, **kwargs):
+        _h5_node = super().__getattribute__('_h5_node')
+        return _h5_node.__getitem__(*args, **kwargs)
+    # TODO as needed add more function like __getitem__ to fully proxy h5_node
+    # or consider using metaclass __getattribute__ for handling special methods
+
+
+class no_compression(dict):
+    """
+    named dict comprehension which which temporarily removes any compression or
+    data filter related arguments from the passed iterable. 
+    """
 
+    # list of keyword parameters to filter
+    __filter_keys__ = {
+        "compression", "shuffle", "compression_opts", "chunks", "fletcher32", "scaleoffset"
+    }
 
-def check_is_hashable(py_obj):
-    """ Check if a python object is hashable
+    def __init__(self, mapping):
+        super().__init__((
+            (key,value)
+            for key,value in ( mapping.items() if isinstance(mapping,dict) else mapping )
+            if key not in no_compression.__filter_keys__
+        ))
+        
 
-    Note: this function is currently not used, but is useful for future
-          development.
+# %% FUNCTION DEFINITIONS
 
-    Args:
-        py_obj: python object to test
+def not_dumpable( py_obj, h_group, name, **kwargs): # pragma: nocover
+    """
+    create_dataset method attached to dummy py_objects which are used to
+    mimic container groups by hickle 4.x. 
+        
+    Raises:
+    -------
+        RuntimeError:
+            in any case as this function shall never be called    
     """
 
-    try:
-        py_obj.__hash__()
-        return True
-    except TypeError:
-        return False
+    raise RuntimeError("types defined by loaders not dumpable")
 
 
-def check_iterable_item_type(iter_obj):
-    """ Check if all items within an iterable are the same type.
+if h5.version.version_tuple[0] >= 3: # pragma: nocover
+    load_str_list_attr_ascii = load_str_list_attr = h5.AttributeManager.get
+    load_str_attr_ascii = load_str_list_attr = h5.AttributeManager.get
 
-    Args:
-        iter_obj: iterable object
+else: # pragma: nocover
```

Comment:
Why should these not be covered by coverage?
One can and should definitely write tests for objects that cannot be hickled.

Reply:
simply cause it is h5py >= 3.0 specific and in testing using h5py 2.10 coverage will compain about not covered like it does on the else clause when testing using h5py == 2.10. In either case one of the two brances will never be hit and thus reported as not covered. As simple as that. In case you have a more appropriate solution telling coverage to not count either branch when runnint tests using other version of h5py please share it with me. The official way i know makes things more complicated here than necessary.